beacon-chain/rpc/validator/proposer.go

```diff
@@ -140,6 +141,103 @@ func (vs *Server) GetBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb
 	return blk, nil
 }
 
+// GetBlockV2 is called by a proposer during its assigned slot to request a block to sign
+// by passing in the slot and the signed randao reveal of the slot. This is used by a validator
+// after the altair fork epoch has been encountered.
+func (vs *Server) GetBlockV2(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BeaconBlockAltair, error) {
+	ctx, span := trace.StartSpan(ctx, "ProposerServer.GetBlock")
+	defer span.End()
+	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
+
+	if vs.SyncChecker.Syncing() {
+		return nil, status.Errorf(codes.Unavailable, "Syncing to latest head, not ready to respond")
+	}
+
+	// Retrieve the parent block as the current head of the canonical chain.
```

Comment:
Most of these inner logic are same as `GetBlock`. Let's refactor the common part to be shared to avoid duplications

One idea is you can call `GetBlock` inside `GetBlockV2`, use the response to construct `BeaconBlockAltair`, Add `SyncAggregate` and compute state root.